app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -162,10 +167,41 @@ def os_disk_caching
       @azure_vm.storage_profile.os_disk.caching
     end
 
+    def image_id
+      image = @azure_vm.storage_profile.image_reference
+      return nil unless image
+      image_uuid = "#{image.publisher}:#{image.offer}:#{image.sku}:#{image.version}"
+      image_uuid
+    end
+
+    def vm_extension
+      return nil unless @azure_vm.resources
+      @vm_extension ||= begin
+        ext_name = @azure_vm.resources.first.id.split('/')[-1]
+        sdk.get_vm_extension(@azure_vm.resource_group, name, ext_name)
+      end
+    end
+
     def script_command
+      unless vm_extension.nil?
+        return @script_command if vm_extension.settings["commandToExecute"].ends_with?("waagent")
+        # Index is based on script_command that is being injected
+        # from the code in #create_vm. It can be partly hard-coded
+        # since the command shall no change frequently.
+        user_cmd_index = (vm_extension.settings["commandToExecute"].index("-c"))+ 4
+        script_command = vm_extension.settings["commandToExecute"][user_cmd_index..-2]
+      else
+        @script_command
+      end
```

Comment:
Here, you have handled only `nil` condition for `vm_extension`. 
I think instead of `unless..else` simply `if..else` would be more readable.

Reply:
done